### DISCUSSION TITLE
Manual revalidate option

### DISCUSSION THREAD

**Discussion by srameshr** [vercel/next.js#25170] [Help] [answered]
### Describe the feature you'd like to request

Make the revalidate option to be on-demand, Instead of setting a fixed number second to re-validate in getStaticProps

### Describe the solution you'd like

It would be great if I could trigger the revalidate request via an exposed API. For example, I would like to manually trigger my NextJS server via an exposed API after an event in my DB is fired.

1. Page: /posts/id-123
2. Post update event trigger
3. My Node API -> ```fetch('my-next-server', { page: '/posts/id-123', revalidate: true })```
4. Manually create a dummy CURL request to /posts/id-123, so it fires a new static regeneration of the page
4. User visits /posts/id-123 and views the updated post.

### Describe alternatives you've considered

N/A

**Comment by srameshr** [comment]
Anything on this?

**Reply by rbatistajs** [reply]
Are working on it
https://github.com/vercel/next.js/discussions/11552#discussioncomment-646963

**Comment by salihbenlalla** [comment]
I was googling this and found this page, it would be nice if a such feature exists in next js, if the post content changed in the database or someone posted a comment on this poste, a revalidation is triggered automatically, without having to set a number of seconds to revalidate, and the user gets a stale data while revalidating ...

**Comment by leerob** [comment, marked as answer]
I'm incredibly excited to share that Next.js 12.1 was just released, which includes **on-demand Incremental Static Regeneration**! We know many of you have been looking forward to this for a while, including myself, and I'm so happy it is here. 

- [Learn about Next.js 12.1](https://nextjs.org/blog/next-12-1)
- [Watch a demo of on-demand ISR](https://youtu.be/BGexHR1tuOA)
- [View another demo and give us your feedback](https://on-demand-isr.vercel.app/)
- [Read the documentation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)

**Reply by mareksuscak** [reply]
@leerob is it possible to issue a revalidate request for a path pattern? Here's the use case: all our dynamic pages fetch snippets of shared data. When shared data changes, we want to invalidate cache for all of these dynamic pages using a single request.
